src/box/lua/config/utils/schema.lua

```diff
@@ -1277,6 +1277,33 @@ local function map_impl(schema, data, f, ctx)
     data = f(data, w, ctx.f_ctx)
 
     if is_scalar(schema) then
+        if schema.type == 'any' then
+            if data == nil then
+                return data
+            end
+            if type(data) ~= 'table' then
+                return data
+            end
+
+            local ok = table_is_array(data)
+            if ok then
+                local res = {}
+                for i, v in ipairs(data) do
+                    walkthrough_enter(ctx, i)
+                    res[i] = map_impl(schema, v, f, ctx)
+                    walkthrough_leave(ctx)
+                end
+                return res
+            end
+
+            local res = {}
+            for k, v in pairs(data) do
+                walkthrough_enter(ctx, k)
+                res[k] = map_impl(schema, v, f, ctx)
+                walkthrough_leave(ctx)
+            end
```

Comment:
Nit: We transform keys for `map` schema nodes. I'm not sure that it is a good idea, TBH, but I would do it here too for consistency.

It adds an interesting ability to assign some value *to the given instance* inside a per-instance mapping:

```yaml
app:
  cfg:
    rpc_priority:
      i-001: 1
      i-002: 2
      i-003: 3
      '{{ instance_name }}': 99
```